src/Perl6/World.nqp

```diff
@@ -530,44 +530,72 @@ class Perl6::World is HLL::World {
           $want
         ) == -1
     }
+
+    method !check-for-PREVIEW ($ver-match, $rev, $default_rev, $rev_mod) {
+        $ver-match.PRECURSOR.worry('PREVIEW modificator is used with released specification 6.' ~ $rev)
+            if nqp::isle_s($rev, $default_rev) && nqp::iseq_s($rev_mod, 'PREVIEW');
+
+        $ver-match.PRECURSOR.worry(
```

Comment:
I think this one should probably be an error.

Reply:
There is revision definition file `PERL6_SPECS` in `tools/templates` now. Looks like:

```
# [*]version [suffix]
# * defines the current spec
c
*d PREVIEW # TODO: It's about time to drop PREVIEW
#e PREVIEW
```

If there is no `PREVIEW` for a revision then this is a error. I'll re-check if this is the case. Otherwise the warning is for the transition period to allow graceful migration.

Reply:
What I think must be done differently here is don't hardcode PREVIEW but rely on what is defined in `PERL6_SPECS`.